SPRacingH7EXTREME - First betaflight STM32H750 400Mhz FC.

hydra opened the pull request:
Target support for SPRacingH7EXTREME.

website: http://seriouslypro.com/spracingh7extreme
shop: https://shop.seriouslypro.com/sp-racing-h7-extreme

NOTE: This PR doesn't enable the second gyro which is mounted at 45 degrees, see https://github.com/betaflight/betaflight/pull/8474

Docteh commented on make/targets_list.mk:
```diff
@@ -12,6 +12,7 @@ OFFICIAL_TARGETS  = \
     SPRACINGF3NEO \
     SPRACINGF4EVO \
     SPRACINGF7DUAL \
+    SPRACINGH7EXTREME \
```
What is this list used for?

hydra replied:
there's a make file goal to build official targets.

Docteh commented on src/main/target/SPRACINGH7EXTREME/SPRACINGH7EXTREME.mk:
```diff
@@ -0,0 +1,13 @@
+            drivers/compass/compass_hmc5883l.c \
+            drivers/barometer/barometer_bmp388.c \
+            drivers/vtx_rtc6705_soft_spi.c
+            
```
On GitHub it's pointing out that the file doesn't end with a newline. Not sure if it's policy or not, but it sticks out.

Also shouldn't this be target.mk? Why is there two .mk files?

hydra replied:
there's two .mk files as the SPRacingH7NANO will be a variant of this, PR to follow this one. Its also why previous PRs from me allowed some build file overrides.

hydra commented:
@mikeller I didn't think the H7 unified target support was ready yet, is it or is there additional work that needs to be done?

Note: There's no unified target for the NUCLEOH743 that was added recently, how was that target was exempt from that requirement?

hydra commented:
Yeah, as noted on the page you linked to there's no H750 or H743 universal target yet, or am I missing something?

https://github.com/betaflight/betaflight/blob/master/unified_targets/docs/CreatingAUnifiedTarget.md#3-flash-the-unified-target-firmware

hydra commented:
I think that adding the universal target for H743 and H750 should be done in separate PR's and not part of this PR.

Suggest merging this as-is until the required unified target has been created, at which point I will be happy to create a new PR for the unified target.

mikeller commented on src/main/target/SPRACINGH7EXTREME/target.c:
```diff
@@ -0,0 +1,58 @@
+/*
```
This is still missing a Unified Target configuration, see https://github.com/betaflight/betaflight#new-requirements-for-the-submission-of-new-and-updated-targets

hydra commented:
ok, after discussion in Slack it was noted that:
* the missing NUCLEOH743 unified target config was an oversight.
* the instructions I was directed at can only be followed up to step 2 at this time.
* the target configuration for the SPRACINGH7EXTREME can be created and added to this PR but cannot be tested yet.

I've added a unified target config to this PR.

LGTM!

jflyper commented:
What's the point of including the unified target config without H743 unified target?

Docteh commented:
It's so that nobody can point at this PR when they want to submit a target without a unified target config.

mikeller commented:
@jflyper: What @Docteh says, plus it's helped us to have Unified Target configurations available during the process of developing Unified Target firmware for the STM32F4 / STM32F7 versions.

jflyper commented:
Ah, so firmware-less config for now.

In that sense, `NUCLEOH743` is missing an unified target config. Do we exempt this target from the guideline as it's a non-flyable dev board?

Btw, we might be ready to add `STM32H7x3` target, as we now have `USE_DMA_SPEC` support. Anyone tried this yet?

mikeller commented:
@jflyper: That was an oversight - I can add a Unified Target configuration for the NUCLEOH743 tonight.

mikeller commented on src/main/target/SPRACINGH7EXTREME/SPRACINGH7EXTREME.mk:
```diff
@@ -0,0 +1,13 @@
+FEATURES       += VCP ONBOARDFLASH SDCARD_SDIO
```
This should go into `target.mk`.

hydra replied:
done

mikeller commented on unified_targets/configs/SPRACINGH7EXTREME.config:
```diff
@@ -0,0 +1,236 @@
+resource MOTOR 6 B07
+resource MOTOR 7 C06
+resource MOTOR 8 C07
+resource SERVO 1 NONE
```
The lines containing `NONE` in here should be removed.

hydra replied:
done
